src/main/java/org/opensearch/security/http/OnBehalfOfAuthenticator.java

```diff
@@ -20,94 +20,130 @@
+            throw new RuntimeException("Unable to find on behalf of authenticator signing key");
+        }
 
         try {
-            if(signingKey == null || signingKey.length() == 0) {
-                log.error("signingKey must not be null or empty. JWT authentication will not work");
-            } else {
-
-                signingKey = signingKey.replace("-----BEGIN PUBLIC KEY-----\n", "");
-                signingKey = signingKey.replace("-----END PUBLIC KEY-----", "");
-
-                byte[] decoded = Decoders.BASE64.decode(signingKey);
-                Key key = null;
-
-                try {
-                    key = getPublicKey(decoded, "RSA");
-                } catch (Exception e) {
-                    log.debug("No public RSA key, try other algos ({})", e.toString());
-                }
-
-                try {
-                    key = getPublicKey(decoded, "EC");
-                } catch (Exception e) {
-                    log.debug("No public ECDSA key, try other algos ({})", e.toString());
-                }
-
-                if(key != null) {
-                    jwtParser = Jwts.parser().setSigningKey(key);
-                } else {
-                    jwtParser = Jwts.parser().setSigningKey(decoded);
-                }
+            final String minimalKeyFormat = signingKey
+                    .replace("-----BEGIN PUBLIC KEY-----\n", "")
+                    .replace("-----END PUBLIC KEY-----", "");
+
+            System.out.println("THIS IS THE minimalKeyFormat of signingKey: " + minimalKeyFormat);
+
+            final byte[] decoded = Base64.getDecoder().decode(minimalKeyFormat);
+            Key key = null;
+
+            try {
+                key = getPublicKey(decoded, "RSA");
+            } catch (Exception e) {
+                log.debug("No public RSA key, try other algos ({})", e.toString());
+            }
 
+            try {
+                key = getPublicKey(decoded, "EC");
+            } catch (Exception e) {
+                log.debug("No public ECDSA key, try other algos ({})", e.toString());
             }
+
+            if (Objects.nonNull(key)) {
+                return Jwts.parser().setSigningKey(key);
+            }
+            // Fallback to the decoded signing key
+            // TODO: Should we ever do this, I think no??
+            return Jwts.parser().setSigningKey(signingKey);
         } catch (Throwable e) {
             log.error("Error while creating JWT authenticator", e);
+            e.printStackTrace();
             throw new RuntimeException(e);
         }
+    }
+
+    private List<String> extractSecurityRolesFromClaims(Claims claims) {
+        Object rolesObject = ObjectUtils.firstNonNull(claims.get("er"), claims.get("dr"));
+        List<String> roles;
+
+        if (rolesObject == null) {
+            log.warn(
+                    "Failed to get roles from JWT claims. Check if this key is correct and available in the JWT payload.");
+            roles = List.of();
```

Comment:
If this message appeared in the logs there is no context to know which token was bad or how to troubleshoot it more.  I'd expect any token like this should be reported as malformed and no more message is needed

Reply:
Fixed~